src/widgets/index/index.ts

```diff
@@ -106,7 +107,7 @@ function resolveScopedResultsFromIndex(widget: Index): ScopedResult[] {
 }
 
 const index = (props: IndexProps): Index => {
-  const { indexName = null } = props || {};
+  const { indexName = null, indexId = indexName } = props || {};
```

Comment:
These types are inferred as `any`. A way to get it typed correctly would be like so:

```ts
// ...

const { indexName = null } = props || ({} as IndexProps);

if (indexName === null) {
  throw new Error(withUsage('The `indexName` option is required.'));
}

// At this point, `indexId` and `indexName` are strings
const { indexId = indexName } = props || ({} as IndexProps);
```